onnxruntime/contrib_ops/cpu/skip_layer_norm.cc

```diff
@@ -105,20 +104,27 @@ Status SkipLayerNorm<T>::Compute(OpKernelContext* p_ctx) const {
   // of the input and skip tensors
   T* skip_input_bias_add_output_data = skip_input_bias_add_output != nullptr ? skip_input_bias_add_output->MutableData<T>() : nullptr;
 
+  const bool skip_broadcasted = (nullptr != skip && skip->Shape()[0] == 1) ? true : false;
+  int skip_size = static_cast<int>(skip_dims[skip_dims_size - 1] * skip_dims[skip_dims_size - 2]);
```

Comment:
We can use one parameter to replace `skip_broadcasted` and `skip_size`:
```
skip_stride = skip->Shape().Size();   # total elements in skip
```

Then use `offset % skip_stride` instead of `(skip_broadcasted ? (offset % skip_size) : offset)`

Reply:
So offset % skip_size (after the edit to skip_size) would account for if the input and skip tensor were the same size and if the skip tensor has a batch size of one?

Reply:
Right. The maximum offset is number of elements in input tensor minus one.
For the case that input and skip having same shape, offset < input_size == skip_size, so offset % skip_size == offset.
You can see that the `offset % skip_size` have same value as `skip_broadcasted ? (offset % skip_size) : offset`